R/modules.R

```diff
@@ -284,7 +320,7 @@ module <- function(label = "module",
       server = server, ui = ui, datanames = datanames,
       server_args = server_args, ui_args = ui_args
     ),
-    class = "teal_module"
+    class = type
```

Comment:
All modules must be of class `teal_module`, other classes may be added.
```suggestion
    class = unique(type, "teal_module")
```

Reply:
I wanted to have it this way but, as mentioned in here https://github.com/insightsengineering/teal/pull/934#discussion_r1369779637
I think we need to rewrite drop/extract_module if we want to have modules with multiple classes

Reply:
When I drafter drop/extract I used a double class module and it worked. It's gonna be fine.

Reply:
I will try rewriting extract/drop_modules so they can handle modules with 2 classes

Reply:
I know, I deleted my messages as I realized they are not valid. I misswatched if with if else : P